gfx-examples/src/shadow.rs

```diff
@@ -641,7 +667,18 @@ impl framework::Example for Example {
     fn resize(&mut self, sc_desc: &wgpu::SwapChainDescriptor, device: &mut wgpu::Device) {
         let mx_total = Self::generate_matrix(sc_desc.width as f32 / sc_desc.height as f32);
         let mx_ref: &[f32; 16] = mx_total.as_ref();
-        self.forward_pass.uniform_buf.set_sub_data(0, framework::cast_slice(&mx_ref[..]));
+        self.forward_pass.uniform_buf.map_write_async(0, 64, |result: wgpu::BufferMapAsyncResult<&mut [f32]>| {
+            if let wgpu::BufferMapAsyncResult::Success(data) = result {
+                data.copy_from_slice(mx_ref);
+            }
+            self.forward_pass.uniform_buf.unmap();
+        });
+
+        {
+            // We need a dummy submit here to trigger the map_write_async to complete
```

Comment:
we should absolutely not do that :)
since we aren't retaining a ring buffer of things, we should write this code synchronously: by creating a new temporary (mapped) buffer, filling it, and issuing a copy operation.